qiskit_experiments/library/characterization/tphi.py

```diff
@@ -0,0 +1,123 @@
+from qiskit_experiments.library.characterization import T1, T2Ramsey
+from qiskit_experiments.library.characterization.analysis.tphi_analysis import TphiAnalysis
+
+
+class Tphi(BatchExperiment):
+    r"""Tphi Experiment Class
+
+    # section: overview
+
+        Tphi is defined as follows:
+
+        :math:`1/T_\phi = 1/T_{2*} - 1/2T_1`.
+
+        For more details, see :py:class:`T1` and :py:class:`T2Ramsey`
+
+    # section: analysis_ref
+        :py:class:`TphiAnalysis`
+
+    # section: reference
+        .. ref_arxiv:: 1 1904.06560v5
+
+    # section: tutorial
+        :doc:`/tutorials/tphi_characterization`
+    """
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            delays_t1 (Iterable[float]): Delay times of the t1 experiment in seconds.
+            delays_t2 (Iterable[float]): Delay times of the t2ramsey experiment in seconds.
+        """
+        options = super()._default_experiment_options()
+        options.delays_t1 = None
+        options.delays_t2 = None
+        return options
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        options = super()._default_transpile_options()
+        return options
+
+    @classmethod
+    def _default_run_options(cls) -> Options:
+        """Default option values for the experiment :meth:`run` method."""
+        options = super()._default_run_options()
+
+        options.shots = 2000
+        return options
+
+    def __init__(
+        self,
+        qubit: int,
+        delays_t1: List[Union[List[float], np.array]],
+        delays_t2: List[Union[List[float], np.array]],
+        osc_freq: float = 0.0,
+        backend: Optional[Backend] = None,
+    ):
+        """Initialize the experiment object.
+
+        Args:
+            qubit: the qubit under test
+            delays_t1: delay times of the T1 experiment
+            delays_t2: delay times of the T2* experiment
+            osc_freq: the oscillation frequency induced using by the user for T2Ramsey
+            backend: Optional, the backend on which to run the experiment
+
+        """
+        self.set_experiment_options = self._default_experiment_options()
+        # Set experiment options
+        self.set_experiment_options.delays_t1 = delays_t1
+        self.set_experiment_options.delays_t2 = delays_t2
+
+        self.exps = []
+        self.exps.append(
+            T1(qubit=qubit, delays=self.set_experiment_options.delays_t1, backend=backend)
+        )
+        self.exps.append(
+            T2Ramsey(
+                qubit=qubit,
+                delays=self.set_experiment_options.delays_t2,
+                backend=backend,
+                osc_freq=osc_freq,
+            )
+        )
+        # Run batch experiments
+        super().__init__(experiments=self.exps, analysis=TphiAnalysis(), backend=backend)
```

Comment:
You can directly override analysis since experiment class has setter. You don't need to modify the signature of the constructor like this.

Reply:
1. I removed the setting of 'shots'
2. Thanks for pointing out the 'analysis' setter. I changed my code to use that instead of adding parameters to the constructors of `BatchExperiment` and `CompositeExperiment`.